### DISCUSSION TITLE
MultipartForm annotation collection support

### DISCUSSION THREAD

**Discussion by Nutrymaco** [resteasy/resteasy#3394] [Ideas]
Should not `@MultipartForm` support mapping in collection fields? Of course technically it is supported, but with only first `InputPart`.

I suggest in case where field type is List/Set in [MultipartFormAnnotationReader](https://github.com/resteasy/resteasy/blob/ce60d03775b7e69865fe48df38f5aa2b464592b9/providers/multipart/src/main/java/org/jboss/resteasy/plugins/providers/multipart/MultipartFormAnnotationReader.java#L150) we should do something like this
```java
if (isCollection(fieldType) && providerFor(fieldType) == null) {
    var collectionData = inputParts.stream()
        .map(part -> part.getBody(...))
        .collect();
    setData(collectionData);
} else {
    InputPart part = list.get(0);
    ...
    data = part.getBody(...);
    ...
    setData(data);
}
```

instead of
```java
InputPart part = list.get(0);
...
data = part.getBody(...);
...
setData(data)
```

If it is reasonable, i can make PR

**Comment by jamezp** [comment]
What version of RESTEasy are you using? Also, could you provide a use-case for this? Even just a simple example method. I'm not quite sure I follow.

**Comment by Nutrymaco** [comment]
Version - 4.7.7.Final (using via Quarkus)
Example:
```java
@POST
@Consumes("multipart/form-data")
public void postForm(@MultipartForm Request request) {

}

class Request {
    @FormParam
    List<Content> content; <-- mapping several files 
}

``` 
And we can get several values by one name via `MultipartFormDataInput`, but can't map several values into one POJO field, because even if we will write custom `MessageBodyReader`, it will be used only for first `inputPart`

**Reply by jamezp** [reply]
In Jakarta REST 3.1 there is a [`EntityPart`](https://jakarta.ee/specifications/restful-ws/3.1/apidocs/jakarta.ws.rs/jakarta/ws/rs/core/entitypart) which would likely replace this.

That said, 4.7 is no longer be developed. Quarkus itself suggests migrating to RESTEasy Reactive and RESTEasy itself has moved to Jakarta REST 3.1+.

**Reply by Nutrymaco** [reply]
I undestand, thanks for answering!
